pyclesperanto_prototype/_tier8/_deskew_y.py

```diff
@@ -41,18 +41,20 @@ def deskew_y(input_image: Image,
 
     from ._AffineTransform3D import AffineTransform3D
     from ._affine_transform import affine_transform
-
+    import math
+    
     # shear in the X plane towards Y
     transform = AffineTransform3D()
     transform.shear_in_x_plane(angle_y_in_degrees=angle_in_degrees)
 
-    # rotate the stack to get proper Z-planes
-    transform.rotate(angle_in_degrees=angle_in_degrees, axis=0)
+    # rotate the stack to get proper Z-planes; rotate 90 - angle around X-axis
+    transform.rotate(angle_in_degrees=90-angle_in_degrees, axis=0)
 
-    # make voxels isotropic, equal to voxel size to raw image in X times scaling_factor.
-    scaling_factor_y = voxel_size_y / voxel_size_x * scaling_factor
-    scaling_factor_z = voxel_size_z / voxel_size_x * scaling_factor
-    transform.scale(scale_x=scaling_factor, scale_y=scaling_factor_y, scale_z=scaling_factor_z)
+    # make voxels isotropic, calculate the new scaling factor for Z after shearing
+    #https://github.com/tlambert03/napari-ndtiffs/blob/092acbd92bfdbf3ecb1eb9c7fc146411ad9e6aae/napari_ndtiffs/affine.py#L57
+    new_dz=math.sin(angle_in_degrees * math.pi/180.0)*voxel_size_z
```

Comment:
Could put this in a function, but didn't want to do it until we've decided on the best approach

Reply:
I think it's ok here. We only need it in this one place, right?

Reply:
Actually, we will need it for the deskew_x function as well. We will change line 56:
scale_factor_z=(new_dz/`voxel_size_y`)*scaling_factor

to 

scale_factor_z=(new_dz/`voxel_size_x`)*scaling_factor

Technically pixel size in x and y are the same..
I'll get a lattice dataset skewed in X direction next week so we can have that example too.